File: leetcode122.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        ls = len(prices)
        if ls == 0:
            return 0
        i = 1
        profit = 0
        local_min = max(prices)
        local_max = 0
        while i < ls:
            if prices[i - 1] >= prices[i]:
                i += 1
                continue
            local_min = prices[i - 1]
            while i < ls and prices[i - 1] < prices[i]:
                i += 1
            local_max = prices[i - 1]
            profit += local_max - local_min

        return profit

File: leetcodec++/leetcode122.cpp
//
// Created by 黄敏辉 on 2019-08-07.
//

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() <= 1)
            return 0;
        int res = 0, low = prices[0], i = 1;
        while(i < prices.size()){
            while(i < prices.size() && prices[i] >= prices[i-1]){
                i++;
            }
            res += prices[i-1] - low;
            if (i < prices.size())
                low = prices[i];
            i++;
        }
        return res;
    }
};